### PR TITLE
[ticket/14849] Add core.acp_extensions_run_action

### PR DESCRIPTION
**Pull request by senky**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-14849

PHPBB3-14849

### CODE REVIEW

File: phpBB/includes/acp/acp_extensions.php [hunk 1]
```diff
@@ -55,6 +56,21 @@ function main()
...
 		$start_time = time();
 
+		/**
+		* Event to run a specific action on extension
+		*
+		* @event core.acp_extensions_run_action
+		* @var	string	action			Action to run
+		* @var	string	u_action		Url we are at
+		* @var	string	ext_name		Extension name from request
+		* @var	string	safe_time_limit	Safe limit of execution time
```
Comment (marc1706) [inline phpBB/includes/acp/acp_extensions.php]: That's an integer

File: phpBB/includes/acp/acp_extensions.php [hunk 2]
```diff
@@ -55,6 +56,21 @@ function main()
...
+		* @var	string	start_time		Start time
```
Comment (marc1706) [inline phpBB/includes/acp/acp_extensions.php]: Also an integer

**Comment by javiexin** [comment]
Sorry to bother, but shouldn't the event be placed just before the switch($action) ?

Checking for the "cancel" action, validating form key, and getting extension metadata in case extension is specified all seem like "required" steps, for any action.  So, whatever extension uses the above event should reimplement all these...

In my opinion, the event would be better placed just before the switch, and would include $md_manager (if possible, as it is an object, and might not be effective to do so... but will most probably require extensions to repeat the code).

**Comment by javiexin** [comment]
@senky One more comment: if the extension sets/changes u_action, this change is lost, and not actioned properly.

**Comment by javiexin** [comment]
New proposal here https://github.com/phpbb/phpbb/pull/4578

**Pull request merged**